### PR TITLE
Migrate utils.dart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/utils.dart [hunk 1]
```diff
@@ -44,16 +42,15 @@ void debugLogger(String message) {
 }
 
 double scaleByFontFactor(double original) {
-  return (original * (ideTheme?.fontSizeFactor ?? 1.0)).roundToDouble();
+  return (original * (ideTheme.fontSizeFactor ?? 1.0)).roundToDouble();
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/utils.dart:45]: make fontsizefactor non-nullable. To do this you can give IdeTheme.fontSize a default value in the constructor `unscaledDefaultFontSize`

File: packages/devtools_app/lib/src/shared/utils.dart [hunk 2]
```diff
@@ -44,16 +42,15 @@ void debugLogger(String message) {
...
 
 bool isDense() {
-  return preferences != null && preferences.denseModeEnabled.value ||
-      isEmbedded();
+  return preferences.denseModeEnabled.value || isEmbedded();
 }
 
 bool isEmbedded() {
-  return ideTheme?.embed ?? false;
+  return ideTheme.embed ?? false;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/utils.dart:53]: make embed non-nullable also. you can give it a default value of false in the constructor

File: packages/devtools_app/lib/src/shared/utils.dart [hunk 3]
```diff
@@ -22,8 +20,8 @@ import 'notifications.dart';
 ///
 /// Shows a `successMessage` [Notification] on the passed in `context`.
 Future<void> copyToClipboard(
-  String data,
-  String successMessage,
+  String? data,
+  String? successMessage,
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/utils.dart:24]: done

File: packages/devtools_app/lib/src/shared/utils.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/utils.dart:45]: done

File: packages/devtools_app/lib/src/shared/utils.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/utils.dart:53]: done

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**